src/pkg/utils/image_utils.go

```
@@ -0,0 +1,192 @@
+	}
+	defer src.Close()
+
+	// Read only the first few bytes to detect format
+	header := make([]byte, 512)
+	_, err = src.Read(header)
+	if err != nil && err != io.EOF {
+		return fmt.Errorf("failed to read file header: %w", err)
+	}
+
+	// Reset to beginning
+	src.Seek(0, 0)
```

golangci-lint (1.64.8)

Error return value of `src.Seek` is not checked

(errcheck)